include/wil/win32_helpers.h

```diff
@@ -20,12 +20,12 @@
 #include <winreg.h>
 #include <objbase.h>
 
+#include "common.h"
+
 // detect std::bit_cast
-#ifdef __has_include
-#if (__cplusplus >= 202002L || _MSVC_LANG >= 202002L) && __has_include(<bit>)
+#if (__WI_LIBCPP_STD_VER >= 20) && WI_HAS_INCLUDE(<bit>, 1) // Assume present if C++20

```

Comment:
Could this be rephrased as `#if WI_HAS_INCLUDE(<bit>, __WI_LIBCPP_STD_VER >= 20)` with equal correctness?

Reply:
The PR description covers why the C++ standard version check is necessary in situations like this. The compiler doesn't use different include paths for different C++ versions, so if your version of the STL supports C++20, but you're compiling as C++17, the `__has_include` check will evaluate to `true` because the header file exists, however if you include it, you will likely get a warning/error saying that the header is unsupported for your target C++ standard version.

Reply:
Ahh.  I misread that part of the PR description.  I did see that part but I didn't understand it well enough.